### ISSUE TITLE
Checking out an unresolvable git ref outputs a horrid error message

### ISSUE THREAD

**Issue by adrienthebo** [puppetlabs/r10k#269]
Given a scenario were r10k is trying to deploy a Git module from an empty repository with no branches, it will provide a pretty dismal error message:

```
[R10K::Action::Puppetfile::Install - ERROR] Could not locate hash at /home/adrien/development/r10k/testing/puppetfile/empty-repository/modules/foo
```

The _hash_ part of the error message is misleading as it's not a hash that can't be located, it's a Git reference that can't be resolved.

r10k should indicate what actually failed in this case and provide a useful error message.

**Comment by adrienthebo** [comment]
See https://tickets.puppetlabs.com/browse/RK-10

**Comment by adrienthebo** [comment]
Merged in 0bc6a95; this will be released in 1.4.1.
